src/pages/en/reference/integrations-reference.md

```diff
@@ -40,30 +39,161 @@ interface AstroIntegration {
 }
 ```
 
-
-<!-- TODO: Detailed Hooks Reference
-
 ## Hooks
 
 ### astro:config:setup
+
+**When it's run:** on initialization, before either the [Vite](https://vitejs.dev/config/) or [Astro config](/en/reference/configuration-reference/) have resolved.
```

Comment:
Nit: can you follow the format that we use on https://deploy-preview-388--astro-docs-2.netlify.app/en/reference/configuration-reference/

```diff
- **When it's run:** ...
- 
- **Use case:** ...
+ **When it's run:** ...  (note: 2 spaces at the end for a newline)
+ **Use case:** ...
```

Also, "When it's run" and "Use case" are almost in two different voices. Stylistically, I like to shorten these so that they are easier for the readers brain to "key" to when scanning the page. Example:



 
```md
**Lifecycle:** ... (alt: "Stage")
**Purpose:** ...
```

Finally, it could be interesting to do something like what Rollup does with their hooks docs (ex: https://rollupjs.org/guide/en/#buildend). This "type" item would actually also solve my comment I made about wanting to see an example with all of the options so that users knew what "config" option was referring to:

```
Type: `({config, command, updateConfig, addRenderer, injectScript}) => void`
Previous Hook: None
Next Hook: `astro:config:done` (clickable link to it)
```

Reply:
I tried lifecycle but it didn't make the most sense in context. I landed on simply "when" and "why!"